### ISSUE TITLE
BUG Issue ( segmentation fault  during backward)

### ISSUE THREAD

**Issue by Feynman1999** [MegEngine/MegEngine#463]
## 环境
1.系统环境：ubantu
2.MegEngine版本： 1.9.0
3.python版本： 3.7

## 复现步骤
简化case:

```
import megengine
import megengine.functional as F
from megengine.autodiff import GradManager
import megengine.module as M
import megengine.optimizer as optim
import numpy as np


class MyModel(M.Module):
     def __init__(self):
         super().__init__()
         self.fc = M.Linear(50, 4)

     def forward(self, input):
         x = self.fc(input)
         return x

model = MyModel()
gm = GradManager().attach(model.parameters())
optimizer = optim.Adam(model.parameters(), lr=0.01)

input =  F.zeros((100, 50))

with gm:
    output = model(input) # (100, 4)
    output = output.reshape(2, 50, 4)

    rgb = output[:, :, :3] # (2, 50, 3)
    weights = output[:, :, 3:4] # (2, 50, 1)
    rgb = F.sum(weights * rgb, axis=-2) # (2, 3)

    # annotation this line is ok
    some_info_of_weights = F.sum(weights, axis=-2)

    target = F.ones((2, 3))
    loss = F.abs(target - rgb).sum()
    gm.backward(loss)
    optimizer.step().clear_grad()

````
## 请提供完整的日志及报错信息
如果加上这句话，some_info_of_weights = F.sum(weights, axis=-2)， 就会直接segmentation fault，没有其他报错信息

1.8.2版本没问题,

**Comment by zrlsdy** [comment]
我来更进一下这个问题。

**Comment by Feynman1999** [comment]
1.9.1好像也会报错

**Comment by zrlsdy** [comment]
出现这个问题，是因为MegEngine改变了求导方式。之前版本的求导是从输出节点，记录反向图，这种情况下`some_info_of_weights`这种没有参与运算的节点，不会进行求导操作。现在的求导，是正向时就记录相关信息，用于构建反向图，`some_info_of_weights`会被记录，从而导致这个问题。所以，这是目前求导方式的缺陷导致的bug，已经记录下来，后续会进行修复。

**Comment by xxr3376** [comment]
Confirmed fixed in v1.10.1
